rasa/core/actions/action.py

```diff
@@ -625,7 +625,14 @@ async def _utter_responses(
         """Use the responses generated by the action endpoint and utter them."""
         bot_messages = []
         for response in responses:
+            generated_response = None
```

Comment:
can we please add a test for this? Should have been caught by a test in the first place.

Reply:
> > I want to ask for your opinion here. I think we should still make the sdk change now, so that the deprecation message would make sense. Any dependencies I need to be aware of? If we push the sdk change first, it'll also make sense to update these
> 
> I agree somewhat. My reasoning behind doing Rasa Open Source first was that this bug blocks users right now while a weird deprecation message isn't blocking for users.
> 
> I'd make the SDK send both (`template` + `response`) keys for now to make the new SDK version keeps on working with older Rasa Open Source versions.
> 
> Depends all a bit on your time. I can do the SDK change tomorrow morning to support you?

I can do it later tonight. :) I think I'm already getting a lot of help from you. 🙏🏼

Reply:
Ok, that would be awesome! 🙌🏻  As reviewer of the PR I should have caught this and we are in this together so always happy to help.

Reply:
Re: the test, I agree with you to add test coverage. This function is not a pure function so it's a bit hard to test. But I'll do my best. I need some time looking through how we do tests here, probably need to mock out a bunch of stuff. Unfortunately, I'm stuck in a bunch of meetings till later in the evening. But I'll come back to this in the evening. 🙏🏼

Reply:
I think you can just copy [this](https://github.com/RasaHQ/rasa/blob/97ae5f1fbf966aece39273928398a3b5d891495a/tests/core/test_actions.py#L241) test and change `"response": "utter_ask_cuisine"` to `"template": "utter_ask_cuisine"`

Reply:
+ a `pytest.warns(...)` block to check for your deprecation warning

Reply:
I'll take a look. Also it *is* actually a pure function, if it doesn't work out I'll just mock stuff.

Reply:
True. It's not public though. Might make sense mock the endpoint and and run the action through the public interface to be closer to the real world and not as reliant on the specific implementation.

Reply:
Yeah the test you sent me was perfect 💯